From 5511ab75fd4c81d20fecc479b6ab08d2978a39bb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1s=20Gonz=C3=A1lez?=
 <tomasagustin.gonzalezorlando@arm.com>
Date: Tue, 14 Mar 2023 11:17:37 +0000
Subject: [PATCH] net: zperf: improve code quality for tcp_received
 function

Currently, in tcp_received function defined in zperf_tcp_receiver
module, the assignment of session->state to STATE_COMPLETED is not
necessary as it is overwritten on the same path to STATE_NULL and
a session is considered free for both STATE_COMPLETED and
STATE_NULL.
Remove the assignment and remove the case because the state is
never reached.
Remove also the STATE_LAST_PACKET_RECEIVED case because this state
is never reached.

Issue-Id: SCM-5823
Signed-off-by: Luca Fancellu <luca.fancellu@arm.com>
Upstream-Status: Backport [main:d61dcf2f1df93de1aacaaa46ce482f885cf755ef]
---
 subsys/net/lib/zperf/zperf_tcp_receiver.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/subsys/net/lib/zperf/zperf_tcp_receiver.c b/subsys/net/lib/zperf/zperf_tcp_receiver.c
index 3dc1f9bbc00..148a1eade3c 100644
--- a/subsys/net/lib/zperf/zperf_tcp_receiver.c
+++ b/subsys/net/lib/zperf/zperf_tcp_receiver.c
@@ -66,8 +66,6 @@ static void tcp_received(const struct sockaddr *addr, size_t datalen)
 	}
 
 	switch (session->state) {
-	case STATE_COMPLETED:
-		break;
 	case STATE_NULL:
 		zperf_reset_session_stats(session);
 		session->start_time = k_uptime_ticks();
@@ -86,8 +84,6 @@ static void tcp_received(const struct sockaddr *addr, size_t datalen)
 		if (datalen == 0) { /* EOF */
 			struct zperf_results results = { 0 };
 
-			session->state = STATE_COMPLETED;
-
 			results.total_len = session->length;
 			results.time_in_us = k_ticks_to_us_ceil32(
 						time - session->start_time);
@@ -99,10 +95,6 @@ static void tcp_received(const struct sockaddr *addr, size_t datalen)
 
 			session->state = STATE_NULL;
 		}
-
-
-		break;
-	case STATE_LAST_PACKET_RECEIVED:
 		break;
 	default:
 		NET_ERR("Unsupported case");
-- 
2.39.1

